peer/core.yaml

```diff
@@ -107,8 +107,6 @@ peer:
     # This case is useful for docker containers.
     addressAutoDetect: false
 
-    # Peer port to accept connections on
-    port:    30303
```

Comment:
I'm a little apprehensive about dropping this.  It seems we do in fact use port 30303:

```
$ git grep peer.port
peer/main.go:   flags.Int("peer-port", 30303, "Port this peer listens to for incoming connections")
peer/main.go:   viper.BindPFlag("peer_port", flags.Lookup("peer-port"))
```

I am not sure if the viper mechanism automatically binds "peer.port" in YAML to "peer-port" in PFlags, but at the very least we can see that this does seem to be a tunable knob and it might be convenient to allow the knob to be set from YAML.  That means we have a few different scenarios:

1. The YAML and PFlags are already wired together and probably should remain that way.
2. The YAML and PFlags are not wired together but should be, we need to fix.
3. This patch does the right thing.

Please investigate further.  If you would like to submit the compose file deletion as a separate PR, I have no problem merging those.  Thanks @kchristidis